src/graph.rs

```diff
@@ -883,6 +883,12 @@ impl<N, E, Ty, Ix> Graph<N, E, Ty, Ix>
         &self.edges
     }
 
+    /// Return the internal arrays, consuming the graph.
+    pub fn destructure(self) -> (Vec<Node<N, Ix>>, Vec<Edge<E, Ix>>)
```

Comment:
I should mention that I wasn't even sure whether `condensation` should consume the graph. An implementation with signature `&Graph<N, E> -> Graph<Vec<&N>, &E>` may be equally interesting (on the other hand, it could be obtained by composing the consuming implementation with a function `&Graph<N, E> -> Graph<&N, &E>`...).